### ISSUE TITLE
dockerng.image_present with 'build' cannot have require/watch

### ISSUE THREAD

**Issue by uppfinnarn** [saltstack/salt#31513]
### Expected Behavior

Specifying `require` or `watch` on a `dockerng.image_present` with a `build:` parameter should make it trigger according to `watch`, and order according to `require`.

### Actual Behavior

Specifying `require` or `watch` along with `build` makes the state crash:

```
An exception occurred in this state: Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/state.py", line 1624, in call
    **cdata['kwargs'])
  File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1491, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/states/dockerng.py", line 2223, in mod_watch
    return image_present(name, **kwargs)
TypeError: image_present() got an unexpected keyword argument '__reqs__'
```

### Steps to Reproduce Issue

```yaml
python-pip:
  pkg.installed

docker-py:
  pip.installed:
    - require:
      - pkg: python-pip

/some/path/Dockerfile:
  file.managed:
    - source: salt://thing/Dockerfile
    - template: jinja
    - require:
      - file: /some/path

local/thing:{{ pillar['thing']['version'] }}:
  dockerng.image_present:
    - build: /some/path
    - force: True
    - require:
      - pip: docker-py
    - watch:
      - file: /some/path/Dockerfile
```

### Versions Report

```
Salt Version:
           Salt: 2015.8.7

Dependency Versions:
         Jinja2: 2.7.3
       M2Crypto: Not Installed
           Mako: 1.0.0
         PyYAML: 3.11
          PyZMQ: 14.4.0
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: 0.8.6
       cherrypy: Not Installed
       dateutil: 2.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: 0.8.2
        timelib: Not Installed

System Versions:
           dist: debian 8.3
        machine: x86_64
        release: 3.16.0-4-amd64
         system: debian 8.3
```

**Comment by jfindlay** [comment]
@uppfinnarn, thanks for reporting.

**Comment by asmaps** [comment]
is there a workaraound for this?

**Comment by uppfinnarn** [comment]
You can use either the old `dockerio` states or a stateful `cmd` state as a workaround.

**Comment by marnovdm** [comment]
I'm also running into this, the workaround with a stateful cmd state works for me, though I'd prefer using the watch/require's directly.

**Comment by elsmorian** [comment]
Am running into this too, was hoping it would be fixed in 2016.3.0 but not so. :-(
